addons/mail/static/src/components/thread_view_topbar/thread_view_topbar.xml

```diff
@@ -49,6 +49,9 @@
                     <t t-if="threadViewTopbar.thread and threadViewTopbar.thread === messaging.starred">
                         <button class="o_ThreadViewTopbar_unstarAllButton btn btn-secondary" t-att-disabled="threadViewTopbar.threadView.messages.length === 0" t-on-click="threadViewTopbar.onClickUnstarAll">Unstar all</button>
                     </t>
+                    <t t-if="messaging.isCurrentUserGuest and isDebugMode">
```

Comment:
Better add a new a field with a expressive name to ThreadViewTopbar that computes the condition.
```suggestion
                    <t t-if="threadViewTopbar.shouldDisplayDebugIcon">
```